frame/alliance/src/lib.rs

```diff
@@ -619,24 +648,80 @@ pub mod pallet {
 		}
 
 		/// Initialize the founders, fellows, and allies.
+		/// Founders must be provided to initialize the Alliance.
+		///
+		/// Provide witness data to disband current Alliance before initializing new.
+		/// Alliance must be empty or disband first to initialize new.
 		///
-		/// This should only be called once, and must be called by the Root origin.
-		#[pallet::weight(T::WeightInfo::init_members(
+		/// Alliance is only disband if new members set is not provided.
+		///
+		/// Must be called by the Root origin.
+		#[pallet::weight(T::WeightInfo::force_set_members(
 			T::MaxFounders::get(),
 			T::MaxFellows::get(),
-			T::MaxAllies::get()
+			T::MaxAllies::get(),
+			witness.proposals,
+			witness.voting_members,
+			witness.ally_members,
 		))]
-		pub fn init_members(
+		pub fn force_set_members(
 			origin: OriginFor<T>,
 			founders: Vec<T::AccountId>,
 			fellows: Vec<T::AccountId>,
 			allies: Vec<T::AccountId>,
+			witness: ForceSetWitness,
 		) -> DispatchResult {
 			ensure_root(origin)?;
 
-			// Cannot be called if the Alliance already has Founders or Fellows.
-			// TODO: Remove check and allow Root to set members at any time.
-			// https://github.com/paritytech/substrate/issues/11928
+			if !witness.is_zero() {
+				// Disband Alliance by removing all members and returning deposits.
+				// Veto and remove all active proposals to avoid any unexpected behavior from
+				// actionable items managed outside of the pallet. Items managed within the pallet,
+				// like `UnscrupulousWebsites`, are left for the new Alliance to clean up or keep.
+
+				ensure!(
+					T::ProposalProvider::proposals_count() <= witness.proposals,
+					Error::<T, I>::BadWitness
+				);
+				ensure!(
+					Self::votable_members_count() <= witness.voting_members,
+					Error::<T, I>::BadWitness
+				);
+				ensure!(
+					Self::ally_members_count() <= witness.ally_members,
+					Error::<T, I>::BadWitness
+				);
+
+				let mut proposals = T::ProposalProvider::proposals();
+				for hash in proposals.iter() {
+					T::ProposalProvider::veto_proposal(*hash);
+				}
+
+				let mut members = Self::votable_members();
+				T::MembershipChanged::change_members_sorted(&[], &members, &[]);
+
+				members.append(&mut Self::members_of(MemberRole::Ally));
+				for member in members.iter() {
+					if let Some(deposit) = DepositOf::<T, I>::take(&member) {
+						let err_amount = T::Currency::unreserve(&member, deposit);
+						debug_assert!(err_amount.is_zero());
+					}
+				}
+
+				Members::<T, I>::remove(&MemberRole::Founder);
+				Members::<T, I>::remove(&MemberRole::Fellow);
+				Members::<T, I>::remove(&MemberRole::Ally);
+
+				members.sort();
+				proposals.sort();
+				Self::deposit_event(Event::AllianceDisband { members, proposals });
+			}
+
+			if founders.is_empty() {
+				ensure!(fellows.is_empty() && allies.is_empty(), Error::<T, I>::FoundersMissing);
+				// new members set not provided.
+				return Ok(())
```

Comment:
I will return an actual weight here, and below. It can vary a lot from the assumed one.